### PR TITLE
Replace Probot labeler with GitHub Actions labeler

### PR DESCRIPTION
**Pull request by ddbeck**
This PR removes the Probot labeler and reinstates the official `@actions/labeler` action using the new `pull_request_target` event. See #6522 for background. It _should_ work this time. 🤞

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: 'Pull Request Labeler'
+on:
+  - pull_request_target
```
Comment (foolip) [inline .github/workflows/labeler.yml:3]: I didn't know about https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target before, seems quite useful!

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
I guess the plan is to land this and see if it works. @ddbeck have you tried running it on a fork of the repository, to shake out any problems? (It has to be a fork because the config will have to be on the default branch. Or maybe it could be done with two branches in this repo, but there's no need to try it here.)

**Comment by schalkneethling** [comment]
@ddbeck This looks good to me. From what I can see and read, this should indeed do the trick.

**Comment by ddbeck** [comment]
Thank you @foolip and @schalkneethling for the reviews! I'm going to deem your reviews sufficient to merge here; I'll be opening a demonstration PR to verify shortly. I'll report progress on #6522.

**Comment by ddbeck** [comment]
Ugh, @foolip sorry I forgot to actually answer your question: yes, I did it test it, but because of the GitHub Actions permissions model and because GitHub won't let you have two forks of the same repo, I had to test it on a copy of this repo and a fork on my account (it worked).

**Comment by foolip** [comment]
Cool, thanks for getting this done!

**Pull request merged**